bijection-hbase/src/main/scala/com/twitter/bijection/hbase/HBaseInjections.scala

```diff
@@ -14,31 +14,103 @@
-  implicit lazy val bigdecimal2BytesInj: Injection[BigDecimal, BigDecimalBytes] = fromBijectionRep[BigDecimal, BigDecimalBytes]
-  implicit lazy val string2BytesWritableInj: Injection[String, ImmutableBytesWritable] = fromBijectionRep[String, ImmutableBytesWritable]
-  implicit lazy val int2BytesWritableInj: Injection[Int, ImmutableBytesWritable] = fromBijectionRep[Int, ImmutableBytesWritable]
-  implicit lazy val long2BytesWritableInj: Injection[Long, ImmutableBytesWritable] = fromBijectionRep[Long, ImmutableBytesWritable]
-  implicit lazy val double2BytesWritableInj: Injection[Double, ImmutableBytesWritable] = fromBijectionRep[Double, ImmutableBytesWritable]
-  implicit lazy val float2BytesWritableInj: Injection[Float, ImmutableBytesWritable] = fromBijectionRep[Float, ImmutableBytesWritable]
-  implicit lazy val short2BytesWritableInj: Injection[Short, ImmutableBytesWritable] = fromBijectionRep[Short, ImmutableBytesWritable]
-  implicit lazy val boolean2BytesWritableInj: Injection[Boolean, ImmutableBytesWritable] = fromBijectionRep[Boolean, ImmutableBytesWritable]
-  implicit lazy val bigDecimal2BytesWritableInj: Injection[BigDecimal, ImmutableBytesWritable] = fromBijectionRep[BigDecimal, ImmutableBytesWritable]
+  // Array[Byte] injections
+  implicit lazy val string2BytesInj: Injection[String, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: String => Array[Byte])(Bytes.toString)
+  implicit lazy val long2BytesInj: Injection[Long, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Long => Array[Byte])(Bytes.toLong)
+  implicit lazy val boolean2BytesInj: Injection[Boolean, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Boolean => Array[Byte])(Bytes.toBoolean)
+  implicit lazy val int2BytesInj: Injection[Int, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Int => Array[Byte])(Bytes.toInt)
+  implicit lazy val float2BytesInj: Injection[Float, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Float => Array[Byte])(Bytes.toFloat)
+  implicit lazy val short2BytesInj: Injection[Short, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Short => Array[Byte])(Bytes.toShort)
+  implicit lazy val double2BytesInj: Injection[Double, Array[Byte]] =
+    buildCatchInvert(Bytes.toBytes: Double => Array[Byte])(Bytes.toDouble)
+  implicit lazy val bigDecimal2BytesInj: Injection[BigDecimal, Array[Byte]] =
+    new AbstractInjection[BigDecimal, Array[Byte]] {
+      override def apply(a: BigDecimal) = Bytes.toBytes(a.underlying)
+      override def invert(b: Array[Byte]) = Try { Bytes.toBigDecimal(b) } map (BigDecimal(_))
+    }
+
+  /**
+   * ImmutableBytesWritable injections avoid copying when possible and use the
+   * slice (offset and length) of the underlying byte array.
+   */
+  implicit lazy val string2BytesWritableInj =
+    new ImmutableBytesWritableInjection[String] {
+      override def invert(b: ImmutableBytesWritable) =
+        Option(Bytes.toString(b.get, b.getOffset, b.getLength))
```

Comment:
Is the whole ```Option(...).toTry``` done to avoid a null check? IMO it would be more efficient to just do 

```
Try{
  if(null check) sys.error(...)
 else result
} 
```